src/main/kotlin/graphics/scenery/CatmullRomSpline.kt

```diff
@@ -0,0 +1,71 @@
+package graphics.scenery.proteins
+
+import cleargl.GLVector
+import kotlin.math.pow
+
+/**
+ * This class offers the logic for creating a CatmuLL Rom Spline. This is essentially
+ * a spline going through control points. For more information see:
+ * https://en.wikipedia.org/wiki/Centripetal_Catmull–Rom_spline
+ * @param controlPoints the list of control points
+ * @param alpha determines the kind of Catmull Rom Spline, set in range of 0..1
+ */
+class CatmullRomSpline(val controlPoints: List<GLVector>, val alpha: Float = 0.5f) {
+    
+    /**
+     * Calculates the parameter t.
+     */
+    fun getT(ti: Float, Pi: GLVector, Pj: GLVector): Float {
+        val exp: Float = (alpha*0.5).toFloat()
+        return(((Pj.x()-Pi.x()).pow(2) + (Pj.y()-Pi.y()).pow(2)
+                + (Pj.z()-Pi.z()).pow(2)).pow(exp) + ti)
+    }
+
+    /**
+     * this function returns the spline points between two points. Please note you need four points
+     * to have a smooth curve.
+     * @param n number of points defining the spline
+     */
+    fun CatmulRomSpline(P0: GLVector, P1: GLVector, P2: GLVector, P3: GLVector, n: Int = 100): List<GLVector> {
```

Comment:
Parameter names should be lowercase :) Is n the number of control points? I would then expect it to appear in the allocation for curvePoints. It however only appears as normalisation factor for t.

Reply:
Nice to know! Well, my description was incorrect: n is in fact the number of points generated between the two control points, e.g. n = 3, then the generated points between (0,0,0) and (1,0,0) would be (0.25,0,0), (0.5,0,0), and (0.75,0,0).